dialogue-java-client/src/main/java/com/palantir/dialogue/HttpChannel.java

```diff
@@ -154,7 +155,13 @@ public int code() {
 
     private static HttpRequest.BodyPublisher toBody(Request request) {
         if (request.body().isPresent()) {
-            return HttpRequest.BodyPublishers.ofInputStream(request.body().get()::content);
+            ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+            try {
+                request.body().get().writeTo(bytes);
+            } catch (IOException e) {
+                throw new SafeRuntimeException("Failed to create a BodyPublisher", e);
+            }
+            return HttpRequest.BodyPublishers.ofByteArray(bytes.toByteArray());
```

Comment:
Here we're able to use the more efficient ofByteArray publisher rather than ofInputStream.

Reply:
Note that before, octet-stream requests would throw (not-implemented) where here they're fully buffered on heap like our feign client. This can be dangerous, but I don't think we should spend the time to optimize that case unless we decide to go all in on the jdk11 client.